Koblicica Andrei Radu/ProblemeL5/3.5L5/main.c

```diff
@@ -0,0 +1,106 @@
+#include <stdio.h>
+#include <stdlib.h>
+int n;
+void addition(int **mat1, int **mat2, int **A, int *n)
+{
+    int i,j;
+    for(i=0; i<*n; i++)
+    {
+        for(j=0; j<*n; j++)
+        {
+            A[i][j]=mat1[i][j]+mat2[i][j];
+        }
+    }
+}
+void substraction(int **mat1, int **mat2, int **A, int *n)
+{
+    int i,j;
+    for(i=0; i<*n; i++)
+    {
+        for(j=0; j<*n; j++)
+        {
+            A[i][j]=mat1[i][j]-mat2[i][j];
+        }
+    }
+}
+/*void multiplication(int **mat1, int **mat2, int **A, int *n)
+{
+    int i,j,a;
+    for(i=0; i<n; i++)
+    {
+        for(j=0; j<n; j++)
+        {
+            A[i][j]=0;
+            for(a=0; a<n; a++)
+            {
+                A[i][j]=A[i][j]+mat1[i][a]*mat2[a][j];
+            }
+        }
+    }
+}*/
+int main()
+{
+    scanf("%d",&n);
+    int i,j;
+    int **mat1=(int*)malloc(sizeof(int*)*n);
+    for(int i=0; i<n; i++)
+    {
+        mat1[i]=(int*)malloc(sizeof(int)*n);
+    }
+    int **mat2=(int*)malloc(sizeof(int*)*n);
+    for(int i=0; i<n; i++)
+    {
+        mat2[i]=(int*)malloc(sizeof(int)*n);
+    }
+    int **A=(int*)malloc(sizeof(int*)*n);
+    for(int i=0; i<n; i++)
+    {
+        A[i]=(int*)malloc(sizeof(int)*n);
+    }
+    for(int i=0; i<n; i++)
```

Comment:
Because you're basically doing this twice, you could also move this matrix read to a function